test/core/event_engine/test_suite/tests/dns_test.cc

```diff
@@ -25,7 +68,747 @@ void InitDNSTests() {}
 }  // namespace experimental
 }  // namespace grpc_event_engine
 
+#ifdef GPR_WINDOWS
 class EventEngineDNSTest : public EventEngineTest {};
 
-// TODO(hork): establish meaningful tests
+// TODO(yijiem): make the test run on Windows
 TEST_F(EventEngineDNSTest, TODO) { grpc_core::ExecCtx exec_ctx; }
+#else
+
+namespace {
+
+using grpc_event_engine::experimental::EventEngine;
+using SRVRecord = EventEngine::DNSResolver::SRVRecord;
+using testing::ElementsAre;
+using testing::Pointwise;
+using testing::UnorderedPointwise;
+
+// TODO(yijiem): make this portable for Windows
+constexpr char kDNSTestRecordGroupsYamlPath[] =
+    "test/core/event_engine/test_suite/tests/dns_test_record_groups.yaml";
+constexpr char kHealthCheckRecordName[] =
+    "health-check-local-dns-server-is-alive.resolver-tests.grpctestingexp";
+
+MATCHER(ResolvedAddressEq, "") {
+  const auto& addr0 = std::get<0>(arg);
+  const auto& addr1 = std::get<1>(arg);
+  return addr0.size() == addr1.size() &&
+         memcmp(addr0.address(), addr1.address(), addr0.size()) == 0;
+}
+
+MATCHER(SRVRecordEq, "") {
+  const auto& arg0 = std::get<0>(arg);
+  const auto& arg1 = std::get<1>(arg);
+  return arg0.host == arg1.host && arg0.port == arg1.port &&
+         arg0.priority == arg1.priority && arg0.weight == arg1.weight;
+}
+
+MATCHER(StatusCodeEq, "") {
+  return std::get<0>(arg).code() == std::get<1>(arg);
+}
+
+// Copied from tcp_socket_utils_test.cc
+// TODO(yijiem): maybe move those into common test util
+EventEngine::ResolvedAddress MakeAddr4(const uint8_t* data, size_t data_len,
```

Comment:
I think the tests would be more readable if these methods took the addresses in URI form instead of an array of bytes.

Reply:
This is a good comment. Done. Silly me spending the time to figure out those raw bytes...